src/plugins/login/azureLoginPlugin.test.ts

```diff
@@ -82,19 +82,22 @@ describe("Login Plugin", () => {
     expect(sls.variables["subscriptionId"]).toEqual("azureSubId");
   });
 
-  it("logs an error from authentication and exits process", async () => {
+  it("logs an error from authentication and crashes with it", async () => {
     unsetServicePrincipalEnvVariables();
-    process.exit = jest.fn() as any;
-    const errorMessage = "This is my error message";
+    const error = new Error("This is my error message")
     AzureLoginService.interactiveLogin = jest.fn(() => {
-      throw new Error(errorMessage);
+      throw error;
     });
     const sls = MockFactory.createTestServerless();
-    await invokeLoginHook(false, sls);
+    try {
+      await invokeLoginHook(false, sls);
+      throw new Error("Unexpected");
```

Comment:
This is only to ensure that test will fail if for some reason `invokeLoginHook` wouldn't crash (see that `expect` at L96 won't be invoked if we do not throw in `try` clause - and not invoking _expect_ means that we do not confirm on that behavior).

Reply:
Right, but I think if that were the case and this error were actually thrown, the `expect` assertion within the cache would fail because it's the wrong error. Because we're substituting the `interactiveLogin` function with an anonymous function that throws an error, it should throw every time. If it doesn't, we have other issues

Reply:
Does this test fail if you remove L94?

Reply:
> Right, but I think if that were the case and this error were actually thrown, the expect assertion within the cache would fail because it's the wrong error.

Exactly that's the intention

> Because we're substituting the interactiveLogin function with an anonymous function that throws an error, it should throw every time. If it doesn't, we have other issues

Yes, and it's exactly why I configured the test to fail, and not to silently pass if unexpectedly `invokeLoginHook` doesn't crash

> Does this test fail if you remove L94?

It wouldn't fail if `invokeLoginHook` would unexpectedly resolve with success (and other constraints will be met) - and that would be the test setup error.

--- 

Anyway I've improved the test by using more straightforward construct, which should make the intention (what we're actually testing) clear.